### PR TITLE
UIU-3225 Integrate action to merge preregistration patron record with existing user record

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://folio-org.atlassian.net/browse/UIU-3225

## Approach
Use the `staging-users` API to merge a staging user with an existing one.

## Screenshot

https://github.com/user-attachments/assets/1467d8fe-94e9-4b85-b416-1dbca0392264

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by alisher-epam — APPROVED** [review]

File: src/views/PatronPreRegistrationRecordsDuplicates/PatronPreRegistrationRecordsDuplicates.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ const PatronPreRegistrationRecordsDuplicates = ({
           style={{ height: wrapperHeight }}
         >
           <PreRegistrationRecordsDuplicatesList
-            isLoading={isLoading || isUsersLoading}
+            isLoading={isLoading || isFetching}
```
Comment (NikitaSedyx) [inline src/views/PatronPreRegistrationRecordsDuplicates/PatronPreRegistrationRecordsDuplicates.js:103]: looks strange isLoading || isFetching, could you please add more reasonable names? isUsersLoading was fine, but what another isLoading in this context is not clear

File: src/views/PatronPreRegistrationRecordsDuplicates/PatronPreRegistrationRecordsDuplicates.js [hunk 2]
```diff
@@ -19,7 +19,7 @@ import { getPatronDuplicatesQuery } from '../../utils';
 import PreRegistrationRecordsDuplicatesList from './PreRegistrationRecordsDuplicatesList';
 
 const PatronPreRegistrationRecordsDuplicates = ({
-  isLoading,
+  isLoading: isLoadingProp,
```
Comment (NikitaSedyx) [inline src/views/PatronPreRegistrationRecordsDuplicates/PatronPreRegistrationRecordsDuplicates.js:22]: isLoadingProp is strange as well :) just call isLoading<What> User? StageUsers? etc

**Review by NikitaSedyx — APPROVED** [review]
apart from comment

File: src/views/PatronPreRegistrationRecordsDuplicates/PatronPreRegistrationRecordsDuplicates.js (hunk 2 above)
Comment (NikitaSedyx) [reply src/views/PatronPreRegistrationRecordsDuplicates/PatronPreRegistrationRecordsDuplicates.js:22]: try to find :) knowing what it does you can give a better name
BTW if isLoading is used to only disable smth, it means that prop shouldn't sound isLoading, it should be smth like smthDisabled/whatever else

**Pull request merged**